### PR TITLE
refactor: internals, lightningcss and overall perf optimizations

### PR DESCRIPTION
**Pull request by astahmer**
## CSS generation
- rm chunks (outdir/chunks/xxx.css)
- sort rules before generating CSS (instead of previous approach of sorting after all CSS has been generated using a postcss plugin)
- insert rules in the ideal order by default, (maybe unlocking the possibility to remove @layer rules ?)
- hash each style objects into unique string to de-duplicate CSS rules and only generate them at most once (instead of previous approach of removing duplicates after all CSS has been generated using a postcss plugin)
- share the Stylesheet context between the different generator engines (global, reset, tokens, parser, static css etc)
- change `optimizeCss` implementation from postcss plugins to lightningcss

## Static CSS
- config.staticCss refactoring + StaticCss engine + static css for patterns and slot recipes
- new RuleProcessor to unify the style rules creation

## Fixes
- fix logger errors, print the stack trace as well instead of swallowing the error entirely
- fix `indeterminate` typo in `@pandacss/preset-base` conditions

## Misc
- refactor internals, move tests in appropriate packages, remove dependencies on `@pandacss/fixture` in each package and unify fixtures, only export the fixturePreset / fixture defaults config, remove some deadcode
- panda ship add a `schemaVersion` in the generated json

## 💣 Is this a breaking change (Yes/No):

no

### CODE REVIEW

**Comment by astahmer** [comment]
https://github.com/chakra-ui/panda/issues/503

File: packages/generator/__tests__/create-anatomy.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
+// https://github.com/chakra-ui/zag/blob/215df8c66e953de02904c8419c1650f344e28faa/packages/anatomy/src/create-anatomy.ts
```
Comment (segunadebayo) [inline packages/generator/__tests__/create-anatomy.ts:1]: Why not install zag-js anatomy for this one?

Comment (astahmer) [reply packages/generator/__tests__/create-anatomy.ts:1]: cause it's just for tests and I didnt want to have the full selectors

https://github.com/chakra-ui/panda/pull/1544/files/ccf4825779a2a6855772a81370cadc496dcc4603..b5d8245cc937744aadcaa6c9a2ceb222ca52bd9b#diff-c836854eabc759fe5cdbff040b25cef0db437818faf43c9ad507b5294546432dR35-R39

**Comment by segunadebayo** [comment]
Moving this to draft since it'll never be merged.

**Comment by segunadebayo** [comment]
Closing this since we only use it as reference

**Pull request closed without merging**